src/main/scala/tsiHost/TsiHostPeriphery.scala

```diff
@@ -0,0 +1,67 @@
+package testchipip
+
+import chisel3._
+import chisel3.util._
+
+import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util._
+import freechips.rocketchip.amba.axi4._
+
+/**
+ * Configuration parameter to configure a set of TSI Host Widgets
+ */
+case object PeripheryTSIHostKey extends Field[Seq[TSIHostParams]](Nil)
+
+/**
+ * Trait to create a set of TSI Host Widgets
+ */
+trait HasPeripheryTSIHostWidget { this: BaseSubsystem =>
+  val tsiHostWidgetNodes = p(PeripheryTSIHostKey).map { ps =>
+    // TODO: Measure BW if using sbus instead of pbus
+    val (hostWidget, hostWidgetMemNode) = TLTSIHostWidget.attach(TSIHostWidgetAttachParams(ps, pbus))
+
+    hostWidget
+  }
+
+  val tsiMemTLNodes = (p(PeripheryTSIHostKey) zip tsiHostWidgetNodes).map { case (params, node) =>
+    val device = new MemoryDevice
+
+    val idBits = 4 // copying the ExtMem
```

Comment:
IIRC these numbers were added so that you could elaborate the IOs for the FPGA DRAM memory similar to how RC does the AXI4 backing memory and TL MMIO port (https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/subsystem/Ports.scala). 

I think these two parameters need to be set like the `MasterPortParams` since they should really be set by the test chip... not by the bring up platform. I'll push a change.

Reply:
Actually on second thought, `idBits` should be specified by the FPGA (since it contains the backing memory) and the `beatBytes` should be specified by the test chip (since the other RC examples do that).